.github/workflows/build.yml

```diff
@@ -0,0 +1,390 @@
+          PROFILE: selenium
+          RAILS_ENV: test
+          SELENIUM_HUB_HOSTNAME: selenium-chrome
+          DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+          CUC_DRIVER: chrome
+          APP_URL: http://school-experience:3000
+
+      - name:  Keep Unit Tests Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: selenium_cucumber_tests
+          path: ${{ github.workspace }}/out/*
+
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    needs: [ selenium_cucumber_tests, cucumber_tests , security_tests, spec_tests ]
+    steps:
+
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: set-up-environment
+        uses: DFE-Digital/github-actions/set-up-environment@master
+
+      - uses: Azure/login@v1
+        with:
+            creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - uses: Azure/get-keyvault-secrets@v1.2
+        id:   azSecret
+        with:
+           keyvault: ${{ secrets.KEY_VAULT}}
+           secrets: 'SONAR-TOKEN'
+
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ steps.azSecret.outputs.SONAR-TOKEN }}
+
+  review:
+    name: Review Deployment Process
+    needs: [  sonarcloud  ]
+    if: github.ref != 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2.3.4
+
+      - name: set-up-environment
+        uses: DFE-Digital/github-actions/set-up-environment@master
+
+      - uses: Azure/login@v1
+        with:
+            creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - uses: Azure/get-keyvault-secrets@v1.2
+        id:   azSecret
+        with:
+           keyvault: ${{ secrets.KEY_VAULT}}
+           secrets: 'ACTIONS-API-ACCESS-TOKEN, SLACK-WEBHOOK'
+
+      - name: Get reference Number
+        run: echo "Event Number - ${{github.event.number}}"
+
+#     - name: Trigger Review Deployment
+#       uses: benc-uk/workflow-dispatch@v1.1
+#       with:
+#         workflow: Deploy to PaaS
+#         token: ${{ steps.azSecret.outputs.ACTIONS-API-ACCESS-TOKEN }}
+#         inputs: '{"environment": "Review", "sha": "${{ github.sha }}" , "pr": "${{github.event.number}}"  }'
+#         ref: ${{ github.event.pull_request.head.ref }}
+
+#     - name: Wait for Deployment to Review
+#       uses: fountainhead/action-wait-for-check@v1.0.0
+#       id: wait-for-review
+#       with:
+#         token: ${{ steps.azSecret.outputs.ACTIONS-API-ACCESS-TOKEN }}
+#         checkName: Deploy Review
+#         ref: ${{ github.event.pull_request.head.ref }}
+
+#     - name: Check if Review Deployment has returned with a failure
+#       if: steps.wait-for-review.outputs.conclusion == 'failure'
+#       run: exit 1
+
+      - name: Post PR comment
```

Comment:
You could also try something like this: https://github.com/DFE-Digital/teaching-vacancies/blob/master/.github/workflows/build_and_deploy.yml#L210
So it ever posts only 1 comment

Reply:
I like that, will use it